TundraTD/Assets/Scripts/Spells/Grimoire.cs

```diff
@@ -2,39 +2,43 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
+using UnityEngine;
 
 namespace Spells
 {
     /// <summary>
     /// A helper class to build spells from elements.
     /// </summary>
-    public static class Grimoire
+    public class Grimoire : MonoBehaviour
     {
-        private static readonly Dictionary<BasicElement, Type> spellTypes;
+        private readonly Dictionary<BasicElement, Type> _spellTypes;
 
-        static Grimoire()
+        [SerializeField] private GameObject[] _spellPrefabs;
+
+        public Grimoire()
         {
-            spellTypes = (from type in Assembly.GetExecutingAssembly().GetTypes()
-                          where typeof(MagicSpell).IsAssignableFrom(type) && !type.IsAbstract
-                          let element = type.GetCustomAttribute<SpellAttribute>()
-                          where element != null
-                          select (element.CoreElement, type)).ToDictionary(x => x.CoreElement, y => y.type);
+            _spellTypes = (from type in Assembly.GetExecutingAssembly().GetTypes()
+                           where typeof(MagicSpell).IsAssignableFrom(type) && !type.IsAbstract
+                           let element = type.GetCustomAttribute<SpellAttribute>()
+                           where element != null
+                           select (element.CoreElement, type)).ToDictionary(x => x.CoreElement, y => y.type);
         }
 
         /// <summary>
         /// Turns elements into a spell if it's available.
         /// </summary>
         /// <param name="elements">Elements list.</param>
         /// <returns>Created spell which is ready to cast.</returns>
```

Comment:
Add a comment about expecting null as a return

Reply:
I suppose that value can't be null because types list doesn't contain null types and I sort them to find where element isn't null. If you want to check possible null references more effective, I think we could enable nullable context in project settings (if it's available with Unity)